docs/DeveloperGuide.md

```diff
@@ -1,38 +1,324 @@
 # Developer Guide
 
-## Acknowledgements
+--------------------------------------------------------------------------------------------------------------------
+# Contents
+* [Acknowledgements](#acknowledgements)
+* [Getting Started For Beginners](#getting-started-for-beginners)
+* [Design](#design)
+  * [Architecture](#architecture)
+  * [UI Component](#ui-component)
+  * [Manager Component](#manager-component)
+  * [Helper Classes](#helper-classes)
+    * [Command Component](#command)
+    * [Parser Component](#parser)
+    * [Storage Component](#storage)
+    * [Validator Component](#validator)
+  * [Asset Classes](#asset-classes)
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+* [Implementation](#implementation)
+  * [Design Considerations](#design-considerations)
+* [Product Scope](#product-scope)
+  * [Target User](#target-user-profile)
+  * [Value Proposition](#value-proposition)
+  * [User Stories](#user-stories)
+  * [Use Cases](#use-cases)
+  * [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+* [Appendix](#appendix)
 
-## Design & implementation
+--------------------------------------------------------------------------------------------------------------------
+## **Acknowledgements**
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+* Design and Structure of Developer Guide referenced from the
+[AB3-Developer Guide](http://se-education.org/addressbook-level3/DeveloperGuide.html).
 
+--------------------------------------------------------------------------------------------------------------------
+## **Getting Started for Beginners**
 
+Refer to the [_User Guide_](UserGuide.md).
+
+--------------------------------------------------------------------------------------------------------------------
+## **Design**
+
+### Architecture
+
+![Architecture Diagram](diagrams/OverallArch.png)
```

Comment:
Really great diagram that gives an overall architecture overview of your program